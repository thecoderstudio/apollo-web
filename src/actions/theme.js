function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
