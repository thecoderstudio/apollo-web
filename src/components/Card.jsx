import styled from 'styled-components';

const Card = styled.div`
  box-shadow: ${(props) => props.theme.boxShadow};
  padding: 40px;
  border-radius: 10px;
  background-color: ${(props) => props.theme.lightBlack};
`;

export default Card;
