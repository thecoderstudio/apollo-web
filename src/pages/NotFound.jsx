import React from 'react';
import styled from 'styled-components';
import media from '../util/media';
import Link from '../components/links/Link';
import OutlinedButton from '../components/buttons/OutlinedButton';
import notFoundImg from '../images/lost.svg';

const Container = styled.div`
  position: fixed;
  left: 50%;
  transform: translate(-50%, 0);
  width: 100%;
  max-width: 1750px;
  height: 100%;
  display: grid;
  grid-template-columns: [img] 1fr [content] 1fr;
  justify-items: center;
  align-items: center;

  ${
    media.phone`
      grid-template-columns: [content] 1fr;
    `
  }
`;

const SupportingImg = styled.img`
  grid-column: img;
  width: 75%;

  ${
    media.phone`
      display: none;
    `
  }
`;

const Content = styled.div`
  display: flex;
  grid-column: content;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  align-items: center;
`;

const Title = styled.h1`
  margin: 0px;
  text-align: center;
  font-size: 20em;
  color: ${props => props.theme.primary};

  ${
    media.phone`
      font-size: 10em;
    `
  }
`;

const SubTitle = styled.h2`
  margin: 0px;
  text-align: center;
  font-size: 2em;
  color: ${props => props.theme.darkGrey};

  ${
    media.phone`
      font-size: 1em;
    `
  }
`;

const StyledButton = styled(OutlinedButton)`
  margin-top: 64px;
  border: 1px solid ${props => props.theme.primary};
`;

export default function NotFound(props) {
  return (
    <Container>
      <SupportingImg src={notFoundImg} />
      <Content>
        <Title>404</Title>
        <SubTitle>Houston, we have a problem</SubTitle>
        <StyledButton><Link to='/'>Reroute</Link></StyledButton>
      </Content>
    </Container>
  );
}
