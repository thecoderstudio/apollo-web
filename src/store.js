import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import storage from 'redux-persist/lib/storage';
import mapTransformer from './transformers/map';
import rootReducer from './reducers/root';

const persistConfig = {
  key: 'root',
  whitelist: ['auth', 'agent'],
  transforms: [mapTransformer({ whitelist: 'agent' })],
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
