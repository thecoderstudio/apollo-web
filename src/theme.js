export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
