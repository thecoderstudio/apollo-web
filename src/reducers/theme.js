import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
