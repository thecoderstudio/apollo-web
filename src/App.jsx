import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import ProtectedRoute from './components/ProtectedRoute';
import NavBar from './components/NavBar';
import Notifications from './components/notifications/Notifications';
import Dashboard from './pages/Dashboard';
import Welcome from './pages/Welcome';
import AgentDetail from './pages/AgentDetail';
import TerminalPage from './pages/TerminalPage';
import Admin from './pages/Admin';
import NotFound from './pages/NotFound';
import { darkTheme } from './theme';

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=B612:wght@400;700&family=Roboto:wght@300;400&display=swap');

  html, body {
    margin: 0px;
    padding: 0px;
    height: 100%;
    width: 100%;
    font-weight: 500;
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.black};
  }

  body {
    font-family: 'Roboto', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'B612', sans-serif;
  }
`;

const Content = styled.div`
  height: 100%;
  width: 100%;
`;

function App(props) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Content>
            {props.authenticated && props.hasChangedPassword && <NavBar />}
            <Switch>
              <ProtectedRoute exact path="/" component={Dashboard} fallbackComponent={Welcome} />
              <ProtectedRoute exact path="/agent/:agentId/shell" component={TerminalPage} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/agent/:agentId" component={AgentDetail} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/admin" component={Admin} fallbackComponent={NotFound} role="admin" />
              <Route component={NotFound} />
            </Switch>
            <Notifications />
            <GlobalStyle />
          </Content>
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
}

export default connect(
  (state) => ({
    authenticated: state.authenticated,
    hasChangedPassword: state.currentUser.hasChangedInitialPassword
  })
)(App);
