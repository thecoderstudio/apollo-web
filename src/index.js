import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
