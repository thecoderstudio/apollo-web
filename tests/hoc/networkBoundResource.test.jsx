import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { connect, Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Map as ImmutableMap } from 'immutable';
import { ThemeProvider } from 'styled-components';
import axios from 'axios';
import { StatusCodes } from 'http-status-codes';
import waitForExpect from 'wait-for-expect';
import Card from '../../src/components/Card';
import NotFound from '../../src/pages/NotFound';
import withNetworkBoundResource from '../../src/hoc/networkBoundResource';
import { darkTheme } from '../../src/theme';

const mockStore = configureStore([]);
jest.mock('axios');

function getComponent(store, dataId = null, spy = jest.fn()) {
  const Component = connect(
    state => ({
      localData: ImmutableMap(state.localData),
      match: { params: state.params }
    })
  )(withNetworkBoundResource(
    Card,
    (localData, params) => localData.get(dataId),
    (params) => `${process.env.APOLLO_HTTP_URL}/test`,
    (data) => spy(data)
  ));


  return renderer.create(
    <Provider store={store}>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Component />
        </BrowserRouter>
      </ThemeProvider>
    </Provider>
  );
}

describe("network bound resource", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      localData: { name: 'test' },
      params: {}
    });
    store.dispatch = jest.fn();
  });

  it("renders loading", () => {
    axios.get.mockResolvedValue({
      status: StatusCodes.OK,
      data: {}
    });

    const tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders not found", async () => {
    axios.get.mockImplementationOnce(() => Promise.reject({
      response: {
        status: StatusCodes.NOT_FOUND,
        statusText: "not found",
        data: {}
      }
    }));
    const component = getComponent(store)
    let tree = component.toJSON();

    await waitForExpect(() => {
      expect(component.root.findAllByType(NotFound)).toHaveLength(1);
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it("renders success from remote", async () => {
    axios.get.mockResolvedValue({
      status: StatusCodes.OK,
      data: {test: true}
    });

    const spy = jest.fn();
    const component = getComponent(store, null, spy)
    let tree = component.toJSON();

    await waitForExpect(() => {
      expect(spy).toHaveBeenCalledWith({ test: true });
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it("renders success from cache", async () => {
    axios.get.mockResolvedValue({
      status: StatusCodes.OK,
      data: {}
    });

    const component = getComponent(store, "name")
    let tree = component.toJSON();

    await waitForExpect(() => {
      expect(component.root.findAllByType(Card)).toHaveLength(1);
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
