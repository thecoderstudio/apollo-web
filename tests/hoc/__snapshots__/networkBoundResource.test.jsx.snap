// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`network bound resource renders loading 1`] = `
.c0 {
  position: fixed;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 100px;
  width: 100px;
}

<div
  aria-busy="true"
  className="c0"
  style={Object {}}
>
  <svg
    aria-label="audio-loading"
    height={80}
    viewBox="0 0 38 38"
    width={80}
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <linearGradient
        id="a"
        x1="8.042%"
        x2="65.682%"
        y1="0%"
        y2="23.865%"
      >
        <stop
          offset="0%"
          stopColor="#02A3EE"
          stopOpacity="0"
        />
        <stop
          offset="63.146%"
          stopColor="#02A3EE"
          stopOpacity=".631"
        />
        <stop
          offset="100%"
          stopColor="#02A3EE"
        />
      </linearGradient>
    </defs>
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        transform="translate(1 1)"
      >
        <path
          d="M36 18c0-9.94-8.06-18-18-18"
          id="Oval-2"
          stroke="#02A3EE"
          strokeWidth="2"
        >
          <animateTransform
            attributeName="transform"
            dur="0.9s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </path>
        <circle
          cx="36"
          cy="18"
          fill="#fff"
          r={1}
        >
          <animateTransform
            attributeName="transform"
            dur="0.9s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </circle>
      </g>
    </g>
  </svg>
</div>
`;

exports[`network bound resource renders not found 1`] = `
.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  pointer-events: inherit;
}

.c6:hover {
  opacity: 0.9;
}

.c0 {
  position: fixed;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  width: 100%;
  max-width: 1750px;
  height: 100%;
  display: grid;
  grid-template-columns: [img] 1fr [content] 1fr;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  grid-column: img;
  width: 75%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-column: content;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 0px;
  text-align: center;
  font-size: 20em;
  color: #02A3EE;
}

.c4 {
  margin: 0px;
  text-align: center;
  font-size: 2em;
  color: #3E3E3E;
}

.c5 {
  max-height: 50px;
  padding: 12px;
  border: none;
  background-color: #4D02EE;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid #4D02EE;
  margin-top: 64px;
  border: 1px solid #02A3EE;
}

.c5:hover {
  opacity: 0.90;
}

.c5:focus {
  outline: none;
}

.c5:active {
  opacity: 0.80;
}

.c5:disabled {
  background-color: #3E3E3E;
}

.c5:disabled:hover {
  opacity: 1;
  cursor: auto;
}

@media (max-width:47.9375em) {
  .c0 {
    grid-template-columns: [content] 1fr;
  }
}

@media (max-width:47.9375em) {
  .c1 {
    display: none;
  }
}

@media (max-width:47.9375em) {
  .c3 {
    font-size: 10em;
  }
}

@media (max-width:47.9375em) {
  .c4 {
    font-size: 1em;
  }
}

<div
  className="c0"
>
  <img
    className="c1"
    src={[Function]}
  />
  <div
    className="c2"
  >
    <h1
      className="c3"
    >
      404
    </h1>
    <h2
      className="c4"
    >
      Houston, we have a problem
    </h2>
    <button
      className="c5"
    >
      <a
        className="c6"
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        Reroute
      </a>
    </button>
  </div>
</div>
`;

exports[`network bound resource renders success from cache 1`] = `
.c0 {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  padding: 40px;
  border-radius: 10px;
  background-color: #1B1B1B;
}

<div
  className="c0"
  data="test"
/>
`;

exports[`network bound resource renders success from remote 1`] = `
.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  pointer-events: inherit;
}

.c6:hover {
  opacity: 0.9;
}

.c0 {
  position: fixed;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  width: 100%;
  max-width: 1750px;
  height: 100%;
  display: grid;
  grid-template-columns: [img] 1fr [content] 1fr;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  grid-column: img;
  width: 75%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-column: content;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 0px;
  text-align: center;
  font-size: 20em;
  color: #02A3EE;
}

.c4 {
  margin: 0px;
  text-align: center;
  font-size: 2em;
  color: #3E3E3E;
}

.c5 {
  max-height: 50px;
  padding: 12px;
  border: none;
  background-color: #4D02EE;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid #4D02EE;
  margin-top: 64px;
  border: 1px solid #02A3EE;
}

.c5:hover {
  opacity: 0.90;
}

.c5:focus {
  outline: none;
}

.c5:active {
  opacity: 0.80;
}

.c5:disabled {
  background-color: #3E3E3E;
}

.c5:disabled:hover {
  opacity: 1;
  cursor: auto;
}

@media (max-width:47.9375em) {
  .c0 {
    grid-template-columns: [content] 1fr;
  }
}

@media (max-width:47.9375em) {
  .c1 {
    display: none;
  }
}

@media (max-width:47.9375em) {
  .c3 {
    font-size: 10em;
  }
}

@media (max-width:47.9375em) {
  .c4 {
    font-size: 1em;
  }
}

<div
  className="c0"
>
  <img
    className="c1"
    src={[Function]}
  />
  <div
    className="c2"
  >
    <h1
      className="c3"
    >
      404
    </h1>
    <h2
      className="c4"
    >
      Houston, we have a problem
    </h2>
    <button
      className="c5"
    >
      <a
        className="c6"
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        Reroute
      </a>
    </button>
  </div>
</div>
`;
